dokka-integration-tests/gradle/src/main/kotlin/org/jetbrains/dokka/it/gradle/junit/TestedVersions.kt

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+package org.jetbrains.dokka.it.gradle.junit
+
+import org.jetbrains.dokka.it.gradle.utils.SemVer
+import org.jetbrains.dokka.it.optionalSystemProperty
+import org.jetbrains.dokka.it.systemProperty
+
+
+/**
+ * The parameters used to run each invocation of a [DokkaGradlePluginTest] test.
+ *
+ * [DokkaGradlePluginTestExtension] generates all possible values, and creates a test invocation
+ * for each valid combination.
+ *
+ * To access the specific values used for a test invocation, add a [TestedVersions]
+ * parameter in the test function.
+ */
+data class TestedVersions(
+    /** Kotlin Gradle Plugin version. */
+    val kgp: SemVer,
+    /** Gradle version. */
+    val gradle: SemVer,
+    /** Dokka Gradle Plugin version. */
+    val dgp: SemVer,
+    /**
+     * The Android Gradle Plugin version being tested.
+     *
+     * The test must be annotated with [TestsAndroidGradlePlugin] for this value to be set.
+     */
+    val agp: SemVer?,
+    /**
+     * The Compose version being tested.
+     *
+     * The test must be annotated with [TestsCompose] for this value to be set.
+     */
+    val composeGradlePlugin: SemVer?,
```

Comment:
nitpick: looking on this, it feels like it’s better just call properies `kotlin`/`dokka`/`android`/`compose`

Reply:
The problem is there's no single 'compose' version. It's very confusing :(

1. Compose compiler plugin hosted by Google (for Kotlin < 2.0.0)
2. Compose compiler plugin hosted by JB
3. Compose Gradle plugin for applying Compose compiler plugin (2)
4. Jetpack Compose (UI framework for Android) as a part of AGP
5. Compose Multiplatform (JB’s UI framework that is using Jetpack Compose on Android) as a Gradle plugin

All these are usually referred as just Compose.